dotnet/src/Extensions/PromptTemplates.Handlebars/Helpers/KernelHelpers/KernelFunctionHelpers.cs

```diff
@@ -190,16 +192,40 @@ private static void ProcessPositionalArguments(KernelFunctionMetadata functionMe
     }
 
     /// <summary>
-    /// Parse the <see cref="FunctionResult"/> into an object, extracting the appropriate value if the return type is <see cref="ChatMessageContent"/>.
+    /// Parse the <see cref="FunctionResult"/> into an object, extracting wrapped content as necessary.
     /// </summary>
     /// <param name="result">Function result.</param>
     /// <returns>Deserialized object</returns>
     private static object? ParseResult(FunctionResult result)
     {
         var resultAsObject = result.GetValue<object?>();
+
+        // Extract content from wrapper types and deserialize as needed.
         if (resultAsObject is ChatMessageContent chatMessageContent)
         {
-            resultAsObject = chatMessageContent.Content;
+            // If the content is null, then something is wrong with response. Throw an exception.
+            if (chatMessageContent.Content == null)
+            {
+                throw new KernelException("ChatMessageContent.Content is null.");
```

Comment:
nit: If this exception will reach user, we need to provide detailed message about issue and possible ways how to resolve it, if there are any.

Reply:
Resolving this comment, looks like we are not throwing an exception now. Thanks!

Reply:
with the revert back to `resultAsObject is RestApiOperationResponse restApiOperationResponse`, I removed this exception.